src/components/MultiSelectCombobox/MultiSelectCombobox.js

```diff
@@ -0,0 +1,256 @@
+        <input
+          autoFocus
+          id={`${entity}-searchbox`}
+          type="text"
+          aria-labelledby={`${entity}-label`}
+          autoComplete="off"
+          aria-autocomplete="list"
+          aria-controls="listbox"
+          aria-activedescendant="listbox"
+          className="search position-static bg-white border-0 width-full height-4 padding-x-1"
+          data-testid="input-search"
+          value={filter}
+          onChange={(e) => onSearchHanlder(e.target.value)}
+          onClick={() => setShowListBox(true)}
+          onKeyDown={(e) => handleKeyDown(e)}
+        />
+        <FontAwesomeIcon
+          icon={faCaretDown}
+          className="pin-right margin-right-1 padding-top-05"
+          onClick={() => setShowListBox(true)}
+        />
+        {showListBox && (
+          <ul
+            aria-multiselectable="true"
+            role="listbox"
+            aria-labelledby={`${entity}-label`}
+            id="listbox"
+            data-testid="multi-select-listbox"
+            tabIndex="-1"
+            className="list-box bg-white display-block height-15 width-full overflow-y-scroll overflow-x-hidden border-top"
+          >
+            {data.length > 0 ? (
+              data.map((item, i) => (
+                <li
+                  key={i}
+                  role="option"
+                  aria-selected={item.selected}
+                  data-id={item.id}
+                  tabIndex="0"
+                  data-label={item.label}
+                  data-testid="multi-select-option"
+                  className={
+                    item.selected
+                      ? "item selected padding-y-1 padding-x-2 border-top-0 display-flex flex-row flex-justify"
+                      : "item padding-y-1 padding-x-2 border-top-0 display-flex flex-row flex-justify"
+                  }
+                  onClick={optionClickHandler}
+                  onKeyDown={(e) => handleKeyDown(e)}
+                >
+                  <span
+                    data-id={item.id}
+                    data-label={item.label}
+                    className="option-label width-mobile"
+                  >
+                    {item.label}
+                  </span>
+                  {item.selected ? (
+                    <FontAwesomeIcon icon={faCheck} color="#005ea2" />
+                  ) : null}
+                </li>
+              ))
+            ) : (
+              <span className="padding-x-2 padding-top-2">
+                No {entity.replace("-", " ")} match your search.
+              </span>
+            )}
+          </ul>
+        )}
+      </div>
+      {window.addEventListener("click", function (e) {
+        const multiSelectComboboxDiv = document.getElementById(
+          `multi-select-combobox-${entity}`
+        );
+        if (
+          multiSelectComboboxDiv &&
+          !multiSelectComboboxDiv.contains(e.target)
+        ) {
+          setShowListBox(false);
+        }
+      })}
```

Comment:
Double check that this isn't a memory leak, if this event listener is being added on every render but not removed it was cause a major performance drag.

Event listeners should be added in effects and removed in their returns.

Reply:
It does indeed and code has been updated to clean up the event.